torch/utils/data/standard_pipes.ipynb

```diff
@@ -0,0 +1,1006 @@
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[0, 1, 2]\n[5, 3, 4]\n[8, 7, 6]\n[9]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10).batch(3).shuffle(cross_shuffle = False)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "To shuffle items on mini-batch level (without changing internal order of items inside mini-batches) use `batch_level` argument"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "[3, 4, 5]\n[9]\n[0, 1, 2]\n[6, 7, 8]\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10).batch(3).shuffle(batch_level = True)\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "## Unbatch\n",
+    "\n",
+    "Function: `unbatch`\n",
+    "\n",
+    "Description: \n",
+    "\n",
+    "Alternatives:\n",
+    "\n",
+    "Arguments:\n",
+    " \n",
+    "Example:"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "4\n9\n6\n5\n7\n8\n2\n3\n1\n0\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10).batch(3).shuffle().unbatch()\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "## Collate\n",
+    "\n",
+    "Function: `collate`\n",
+    "\n",
+    "Description: \n",
+    "\n",
+    "Alternatives:\n",
+    "\n",
+    "Arguments:\n",
+    " \n",
+    "Example:"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "tensor([0, 1, 2])\ntensor([3, 4, 5])\ntensor([6, 7, 8])\ntensor([9])\n"
+     ]
+    }
+   ],
+   "source": [
+    "dp = ExampleIterPipe(10).batch(3).collate()\n",
+    "for i in dp:\n",
+    "    print(i)"
+   ]
+  },
+  {
+   "source": [
+    "## GroupBy\n",
+    "\n",
+    "Function: `groupby`\n",
+    "\n",
+    "Usage: `dp.groupby(lambda x: x[0])`\n",
+    "\n",
+    "Description: Batching items by combining items with same key into same batch \n",
+    "\n",
+    "Arguments:\n",
+    " - `group_key_fn`\n",
+    " - `group_size` - yeild resulted group as soon as `group_size` elements accumulated\n",
+    " - `guaranteed_group_size:int = None`\n",
+    " - `batch_level`\n",
+    "\n",
+    "#### Attention\n",
+    "As datasteam can be arbitrary large, grouping is done on best effort basis and there is no guarantee that same key will never present in the different groups."
```

Comment:
nit: no guarantee that same key will present

Reply:
Need to make sure:
`buffer_size >= group_size >= guaranteed_group_size`

Reply:
`buffer_size` is not listed in `Arguments`?